portfolio/src/main/webapp/script.js

```diff
@@ -35,11 +35,65 @@ function addRandomFact() {
     
     //remove this fact and the null from the array
     facts.splice(factIndex,1);
-    console.log(facts.toString());
-    console.log(facts.length);
 
     //update i 
     i = facts.length-1;
   }
+}
+
+
 
+// function to calculate the result of the survey
+function tabulateAnswers() {
+  // initialize variables for each choice's score
+  // If you add more choices and outcomes, you must add another variable here.
+  var c1score = 0;
```

Comment:
What can we store these in that would allow you to collect multiples of the same item type?  That would help with not having to update if new choices are added.

Reply:
I'm a bit confused. Are you suggesting using creating an class/object setup?

Reply:
Or perhaps an array?

Reply:
Yep array or map would be a good starting point, eg:
```
let score_totals = {};
for (...) {
  score_totals[choices[i].value]++;
}

let max_score = -1;
let top_category = "";
for (/*iterate through scores_total here*/) {
  if (max_score < score_totals[key]) {
    max_score = score_totals[key];
    top_category = key;
  }
}
```

You can also take it up a notch and build an array/map of objects that holds both the total score and the answerbox.innerHTML message, eg:
```
let categories = {
  c1: {
    total_score: 0,
    message = "Go Surfing!"
  },
  c2: {
    total_score: 0,
    message = "Go Kayaking!"
  },
  c3: {
    total_score: 0,
    message = "Go Hiking!"
  },
  c4: {
    total_score: 0,
    message = "Have a Picnic!"
  }
};
for (...) {
  categories[choices[i].value].total_score++;
}

let max_score = -1;
let top_category = "";
for (/*iterate through categories here*/) {
  if (max_score < categories[key].total_score) {
    max_score = categories[key].total_score
    top_category = key;
  }
}

answerbox.innerHTML = categories[top_category].message;
```